eglib/src/gmisc-unity.c

```diff
@@ -1,10 +1,64 @@
-#ifdef PLATFORM_UNITY
-
 #include <stdlib.h>
 #include <glib.h>
 
+#include "Environment-c-api.h"
+#include "Locale-c-api.h"
 #include "Path-c-api.h"
 
+const gchar *
+g_getenv(const gchar *variable)
+{
+    return UnityPalGetEnvironmentVariable(variable);
+}
+
+gboolean
+g_setenv(const gchar *variable, const gchar *value, gboolean overwrite)
+{
+    // This method assumes overwrite is always true.
+    UnityPalSetEnvironmentVariable(variable, value);
+
+    // No code in Mono actually checks the return value.
+    return TRUE;
+}
+
+void
+g_unsetenv(const gchar *variable)
+{
+    UnityPalSetEnvironmentVariable(variable, "");
```

Comment:
I don't think so. This is what the Mono Windows implementation does. The Posix implementation calls `unsetenv` instead. I wanted to match the Windows implementation since our API more closely mirrors it.